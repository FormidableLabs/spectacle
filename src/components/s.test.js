import S from './s';
import React from 'react';
import { mount } from 'enzyme';

describe('<S />', () => {
  test('should underline text when specified', () => {
    const context = {
      styles: {
        components: {
          s: {
            strikethrough: { color: '#ff0' }
          }
        }
      }
    };
    const wrapper = mount(<S type="strikethrough">Donâ€™t read this!</S>, {
      context
    });
    expect(wrapper).toMatchSnapshot();
  });

  test('should bold text when specified', () => {
    const context = {
      styles: {
        components: {
          s: {
            bold: { color: '#ff0' }
          }
        }
      }
    };
    const wrapper = mount(<S type="bold">You should read this!</S>, {
      context
    });
    expect(wrapper).toMatchSnapshot();
  });

  test('should underline text when specified', () => {
    const context = {
      styles: {
        components: {
          s: {
            underline: { color: '#ff0' }
          }
        }
      }
    };
    const wrapper = mount(<S type="underline">This text is underlined!</S>, {
      context
    });
    expect(wrapper).toMatchSnapshot();
  });

  test('should italicize text when specified', () => {
    const context = {
      styles: {
        components: {
          s: {
            italic: { color: '#ff0' }
          }
        }
      }
    };
    const wrapper = mount(<S type="italic">This text is italicized!</S>, {
      context
    });
    expect(wrapper).toMatchSnapshot();
  });
});
