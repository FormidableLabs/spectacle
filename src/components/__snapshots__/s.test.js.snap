// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<S /> should bold text when specified 1`] = `
<S
  type="bold"
>
  <Styled(span)
    className={undefined}
    styles={
      Object {
        "base": Object {},
        "context": Object {
          "color": "#ff0",
        },
        "user": undefined,
      }
    }
    type="bold"
  >
    <span
      className="css-1oz8dr5"
      type="bold"
    >
      You should read this!
    </span>
  </Styled(span)>
</S>
`;

exports[`<S /> should italicize text when specified 1`] = `
<S
  type="italic"
>
  <Styled(span)
    className={undefined}
    styles={
      Object {
        "base": Object {},
        "context": Object {
          "color": "#ff0",
        },
        "user": undefined,
      }
    }
    type="italic"
  >
    <span
      className="css-1cx0df8"
      type="italic"
    >
      This text is italicized!
    </span>
  </Styled(span)>
</S>
`;

exports[`<S /> should underline text when specified 1`] = `
<S
  type="strikethrough"
>
  <Styled(span)
    className={undefined}
    styles={
      Object {
        "base": Object {},
        "context": Object {
          "color": "#ff0",
        },
        "user": undefined,
      }
    }
    type="strikethrough"
  >
    <span
      className="css-1vlogow"
      type="strikethrough"
    >
      Donâ€™t read this!
    </span>
  </Styled(span)>
</S>
`;

exports[`<S /> should underline text when specified 2`] = `
<S
  type="underline"
>
  <Styled(span)
    className={undefined}
    styles={
      Object {
        "base": Object {},
        "context": Object {
          "color": "#ff0",
        },
        "user": undefined,
      }
    }
    type="underline"
  >
    <span
      className="css-xt7dpi"
      type="underline"
    >
      This text is underlined!
    </span>
  </Styled(span)>
</S>
`;
